ako-infra/avirest/handle_netinfo.go

```diff
@@ -0,0 +1,250 @@
+
+	specJSON, found, err := unstructured.NestedStringMap(crd.UnstructuredContent(), "topology")
+	if err != nil || !found {
+		utils.AviLog.Warnf("NetworkInfo topology not found: %+v", err)
+		return
+	}
+	lr := specJSON["gatewayPath"]
+	ls := specJSON["aviSegmentPath"]
+	utils.AviLog.Infof("key: %s, Adding LR %s, LS %s from networkinfo CR", objKey, lr, ls)
+	client := avicache.SharedAVIClients().AviClient[0]
+	found, cloudModel := getAviCloudFromCache(client, utils.CloudName)
+	if !found {
+		utils.AviLog.Warnf("key: %s, Failed to get Cloud data from cache", objKey)
+		return
+	}
+	updateRequired, lslrList := addSegmentInCloud(cloudModel.NsxtConfiguration.DataNetworkConfig.Tier1SegmentConfig.Manual.Tier1Lrs, lr, ls)
+	if updateRequired {
+		cloudModel.NsxtConfiguration.DataNetworkConfig.Tier1SegmentConfig.Manual.Tier1Lrs = lslrList
+		path := "/api/cloud/" + *cloudModel.UUID
+		restOp := utils.RestOp{ObjName: utils.CloudName, Path: path, Method: utils.RestPut, Obj: &cloudModel,
+			Tenant: "admin", Model: "cloud", Version: utils.CtrlVersion}
+		executeRestOp(objKey, client, &restOp)
+	} else {
+		utils.AviLog.Infof("key: %s, LSLR update not required in cloud: %s", objKey, utils.CloudName)
+	}
+}
+
+func DeleteSegment(obj interface{}) {
+	objKey := "Netinfo" + utils.ObjKey(obj)
+	utils.AviLog.Debugf("key:%s, Network Info DELETE Event", objKey)
+	crd := obj.(*unstructured.Unstructured)
+
+	specJSON, found, err := unstructured.NestedStringMap(crd.UnstructuredContent(), "topology")
+	if err != nil || !found {
+		utils.AviLog.Warnf("key: %s, NetworkInfo topology not found: %v", objKey, err)
+		return
+	}
+	lr := specJSON["gatewayPath"]
+	ls := specJSON["aviSegmentPath"]
+	utils.AviLog.Infof("key: %s, Network Info CR deleted, removing LR %s, LS %s from cloud", objKey, lr, ls)
+	client := avicache.SharedAVIClients().AviClient[0]
+	found, cloudModel := getAviCloudFromCache(client, utils.CloudName)
+	if !found {
+		utils.AviLog.Warnf("key: %s, Failed to get Cloud data from cache", objKey)
+		return
+	}
+	updateRequired, lslrList := delSegmentInCloud(cloudModel.NsxtConfiguration.DataNetworkConfig.Tier1SegmentConfig.Manual.Tier1Lrs, lr, ls)
+	if updateRequired {
+		cloudModel.NsxtConfiguration.DataNetworkConfig.Tier1SegmentConfig.Manual.Tier1Lrs = lslrList
+		path := "/api/cloud/" + *cloudModel.UUID
+		restOp := utils.RestOp{ObjName: utils.CloudName, Path: path, Method: utils.RestPut, Obj: &cloudModel,
+			Tenant: "admin", Model: "cloud", Version: utils.CtrlVersion}
+		executeRestOp(objKey, client, &restOp)
+	} else {
+		utils.AviLog.Infof("key: %s, LSLR update not required in cloud: %s", objKey, utils.CloudName)
+	}
+}
+
+func addSegmentInCloud(lslrList []*models.Tier1LogicalRouterInfo, lr, ls string) (bool, []*models.Tier1LogicalRouterInfo) {
+	for i := range lslrList {
+		if *lslrList[i].SegmentID == ls {
+			if *lslrList[i].Tier1LrID == lr {
+				return false, lslrList
+			}
+			lslrList = append(lslrList[:i], lslrList[i+1:]...)
```

Comment:
if we don't hit the `if` block then we will remove the entry from the `lslrlist`? Why would we do that?

Reply:
As we take out entry from lslrList should we "break" for loop? This will save few cycles. I assume there will not be multiple entries for same SegID.

Reply:
yeah, we can break. added.